knowledge-base/common-onchange-fires-twice.md

```diff
@@ -24,28 +24,35 @@ res_type: kb
 
 ## Description
 
-I observe twice firing onchange event in dropdownlist or other inputs.
+I observe twice firing `OnChange` event in the DropDownList or other inputs.
 
 I want the event to fire only once when the user selects something.
 
+I want to execute my business logic in the `OnChange` event handler only once per value change.
+
 ## Possible Cause
 
-The OnChange event is a user confirmation event - it fires when the user chooses an item from the dropdown list, and also when the user blurs an input (the dropdownlist is, in essence, an input).
+The `OnChange` event is a user confirmation event - it fires when the user chooses an item from the DropDownList popup, and also when the user blurs an input (the DropDownList is, in essence, an input).
 
 For example, pressing Enter in an input will fire the event, but will not remove the focus from the input. Thus, the next click on the page (on a button, another component) will fire the event again.
 
 ## Solution
 
->tip As of `2.22.0`, you can use the `OnBlur` event of the components to capture the `focusout` event, so you may not need to use `OnChange`.
+1. Create an additional variable.
+1. In the `OnChange` event handler:
+    * Get the user input from the `OnChange` event argument or from the `Value` parameter.
```

Comment:
Are these actions supposed to be executed in a particular sequence? If yes, this needs to be a numbered list.